flutter/news/lib/core/http_client.dart

```diff
@@ -0,0 +1,15 @@
+import 'package:dio/dio.dart';
+import 'package:dio_logging_interceptor/dio_logging_interceptor.dart';
+import 'package:flutter_dotenv/flutter_dotenv.dart';
+
+Dio createDio() {
```

Comment:
Given that this `createDio` is only going to be usable with `NewsApiClient`, does it make sense to have it in a separate file? Perhaps defining it in `news_api_client.dart` makes sense

Reply:
Perhaps something like this could make sense, because the dio created here makes sense only for `NewsApiClient`, could we use...

```

@RestApi(baseUrl: "https://newsapi.org")
abstract class NewsApiClient {
  factory NewsApiClient(Dio dio, {String baseUrl}) = _NewsApiClient;
  factory NewsApiClient.create() => NewsApiClient(_createDio());

  @GET("/v2/top-headlines")
  Future<BaseNewsResponse> topHeadLines(@Query("country") String country);
}

Dio _createDio() {
  final dio = Dio();
  dio.options.headers['X-Api-Key'] = dotenv.env['NEWS_API_KEY'];
  dio.interceptors.add(
    DioLoggingInterceptor(
      level: Level.body,
      compact: false,
    ),
  );
  return dio;
}
```

Reply:
Then on the `get-it` implementation, we create the `NewsApiClient` with `NewsApiClient.create()` while keeping the other constructor, that could be useful for tests